test/e2e/router_status_test.go

```diff
@@ -0,0 +1,159 @@
+	if err := waitForRouteStatusClear(t, kclient, routeName, ic.Name); err != nil {
+		t.Fatalf("failed to observe route status clear when ingress controller was deleted: %v", err)
+	}
+}
+
+// TestIngressControllerSelectorUpdateShouldClearRouteStatus modifies an ingress controller's selectors to
+// unadmit a route and verifies it clears the admitted route's status.
+func TestIngressControllerSelectorUpdateShouldClearRouteStatus(t *testing.T) {
+	// Create an ingress controller that can admit our route.
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "ic-selector-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newLoadBalancerController(icName, domain)
+	ic.Spec.RouteSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"type": icName.Name,
+		},
+	}
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	routeName := types.NamespacedName{Namespace: "default", Name: "route" + icName.Name}
+	route := newRouteWithLabel(routeName, icName.Name)
+
+	// Create a route to be admitted by this ingress controller.
+	if err := kclient.Create(context.TODO(), route); err != nil {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), route); err != nil {
+			t.Fatalf("failed to delete router %s: %v", routeName.Name, err)
+		}
+	}()
+
+	// Wait for route to be admitted.
+	admittedCondition := routev1.RouteIngressCondition{Type: routev1.RouteAdmitted, Status: corev1.ConditionTrue}
+	if err := waitForRouteIngressConditions(t, kclient, routeName, ic.Name, admittedCondition); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Update the ingress controller to not select the route to be admitted.
+	if err := kclient.Get(context.TODO(), icName, ic); err != nil {
+		t.Fatalf("failed to get ingress resource: %v", err)
+	}
+	ic.Spec.RouteSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"type": "foo",
+		},
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	// Wait until route status is cleared.
+	if err := waitForRouteStatusClear(t, kclient, routeName, ic.Name); err != nil {
+		t.Fatalf("failed to observe route status clear when ingress controller selector was updated: %v", err)
+	}
+}
+
+// waitForRouteStatusClear waits for route to be unadmitted (e.g. admitted status cleared) by given ingress controller.
+func waitForRouteStatusClear(t *testing.T, cl client.Client, routeName types.NamespacedName, routerName string) error {
+	t.Helper()
+	return wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		route := &routev1.Route{}
+		if err := cl.Get(context.TODO(), routeName, route); err != nil {
+			t.Logf("failed to get route %s: %v", routeName.Name, err)
```

Comment:
Might as well log the whole route name:
```suggestion
			t.Logf("failed to get route %s: %v", routeName, err)
```

Reply:
Done